charts/dns/templates/pdns-ingress.yaml

```diff
@@ -1,23 +1,34 @@
-{{- if .Values.pdns.httpService.enabled }}
-{{- if .Values.pdns.ingress.enabled }}
-{{- $fullName := include "pdns.fullname" . -}}
+{{- if and .Values.service.http.enabled .Values.ingress.enabled -}}
+{{- $fullName := include "stateless-dns.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```

Comment:
I would honestly get rid of this. 1.19 is prehistoric as of 2023.

Reply:
Fair enough. Removed.